src/init/AppRunner.ts

```diff
@@ -57,96 +50,80 @@ export class AppRunner {
     // Parse the command-line arguments
     // eslint-disable-next-line no-sync
     const params = yargs(argv.slice(2))
-      .strict()
       .usage('node ./bin/server.js [args]')
-      .check((args): boolean => {
-        if (args._.length > 0) {
-          throw new Error(`Unsupported positional arguments: "${args._.join('", "')}"`);
-        }
-        for (const key of Object.keys(args)) {
-          // We have no options that allow for arrays
-          const val = args[key];
-          if (key !== '_' && Array.isArray(val)) {
-            throw new Error(`Multiple values were provided for: "${key}": "${val.join('", "')}"`);
-          }
-        }
-        return true;
-      })
-      .options({
-        baseUrl: { type: 'string', alias: 'b', requiresArg: true },
-        config: { type: 'string', alias: 'c', default: defaultConfig, requiresArg: true },
-        loggingLevel: { type: 'string', alias: 'l', default: 'info', requiresArg: true },
-        mainModulePath: { type: 'string', alias: 'm', requiresArg: true },
-        port: { type: 'number', alias: 'p', default: 3000, requiresArg: true },
-        rootFilePath: { type: 'string', alias: 'f', default: './', requiresArg: true },
-        showStackTrace: { type: 'boolean', alias: 't', default: false },
-        sparqlEndpoint: { type: 'string', alias: 's', requiresArg: true },
-        podConfigJson: { type: 'string', default: './pod-config.json', requiresArg: true },
-      })
+      .options(baseYargsOptions)
       .parseSync();
 
     // Gather settings for instantiating the server
-    const loaderProperties: IComponentsManagerBuilderOptions<App> = {
-      mainModulePath: resolveAssetPath(params.mainModulePath),
+    const loaderProperties = {
+      mainModulePath: resolveAssetPath(params.mainModulePath as string),
       dumpErrorState: true,
       logLevel: params.loggingLevel as LogLevel,
     };
```

Comment:
That check for string type will be performed by yargs itself, as we defined it to be string and required in base-options. That type co-ercion is for typescript, as yargs won't do it if we don't pass arguments inline.

Reply:
But by using that type guard if-statement typescript will also be able to deduce that `mainModulePath` is a string. Preferably we should never have casts in the code and always type checks from which the type can be deduced. In case `mainModulePath` is a number or random object here, the `resolveAssetPath` call is going to throw some sort of error about the input object not having a `startsWith` function, while by doing the check first as we do here there is going to be a clearer message.

Reply:
That part of code in `if` block will be never executed because, if it is not string yargs will any-way through the error. because we declared for it. At that point it is sure that `mainModulePath` will be string. Just that typescript cannot infer it. `as` is there for precisely this purpose. When typescript cannot infer, but we know something for sure.

Reply:
That makes sense. I wonder if there is perhaps someone to make it so the yargs call returns a type guard but that might be too complicated. I would add clear comments here explaining why the casts are necessary.

Reply:
Yes, i tried a bit a ts wizardry, but failed

Reply:
It should be fixed now, by using yargs' builder pattern and it's InferredOptions mechanism.